src/include/buffer.h

```diff
@@ -1230,41 +1230,37 @@ inline namespace v15_2_0 {
     }
   };
 
-inline bool operator>(bufferlist& l, bufferlist& r) {
-  for (unsigned p = 0; ; p++) {
-    if (l.length() > p && r.length() == p) return true;
-    if (l.length() == p) return false;
-    if (l[p] > r[p]) return true;
-    if (l[p] < r[p]) return false;
-  }
+inline bool operator==(const bufferlist &lhs, const bufferlist &rhs) {
+  if (lhs.length() != rhs.length())
+    return false;
+  return std::equal(lhs.begin(), lhs.end(), rhs.begin(), rhs.end());
```

Comment:
```suggestion
  return std::equal(lhs.begin(), lhs.end(), rhs.begin());
```

```
2020-04-10T11:33:39.972 INFO:tasks.workunit.client.0.smithi103.stderr:In file included from /usr/include/c++/7/memory:62:0,
2020-04-10T11:33:39.972 INFO:tasks.workunit.client.0.smithi103.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/include/rocksdb/utilities/env_librados.h:8,
2020-04-10T11:33:39.973 INFO:tasks.workunit.client.0.smithi103.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/utilities/env_librados.cc:4:
2020-04-10T11:33:39.973 INFO:tasks.workunit.client.0.smithi103.stderr:/usr/include/c++/7/bits/stl_algobase.h: In instantiation of 'bool std::equal(_IIter1, _IIter1, _IIter2, _BinaryPredicate) [with _IIter
1 = ceph::buffer::v15_2_0::list::iterator_impl<true>; _IIter2 = ceph::buffer::v15_2_0::list::iterator_impl<true>; _BinaryPredicate = ceph::buffer::v15_2_0::list::iterator_impl<true>]':
2020-04-10T11:33:39.973 INFO:tasks.workunit.client.0.smithi103.stderr:/usr/include/rados/buffer.h:1236:67:   required from here
2020-04-10T11:33:39.973 INFO:tasks.workunit.client.0.smithi103.stderr:/usr/include/c++/7/bits/stl_algobase.h:1082:25: error: no match for call to '(ceph::buffer::v15_2_0::list::iterator_impl<true>) (char,
 char)'
2020-04-10T11:33:39.974 INFO:tasks.workunit.client.0.smithi103.stderr:  if (!bool(__binary_pred(*__first1, *__first2)))
2020-04-10T11:33:39.974 INFO:tasks.workunit.client.0.smithi103.stderr:            ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
2020-04-10T11:33:40.083 INFO:tasks.workunit.client.0.smithi103.stdout:CMakeFiles/rocksdb-shared.dir/build.make:6206: recipe for target 'CMakeFiles/rocksdb-shared.dir/utilities/env_librados.cc.o' failed
2020-04-10T11:33:40.084 INFO:tasks.workunit.client.0.smithi103.stderr:make[3]: *** [CMakeFiles/rocksdb-shared.dir/utilities/env_librados.cc.o] Error 1
```
seems GCC7 was trying to use the iterator as a `BinaryPredicate`. as an alternative fix, we should install devtools-9 for a more standard compliant GCC for using librados.

please note, we probably are trying to convince our users to upgrade their compiler for building their librados applications with librados from Pacific if we want go with this change.

see http://pulpito.ceph.com/kchai-2020-04-10_10:07:46-rados-wip-kefu-testing-2020-04-10-1430-distro-basic-smithi/4942532/

Reply:
applied, thanks!